### ISSUE TITLE
Reprocessing of complete queue in case of start_from_beginning: true

### ISSUE THREAD

**Issue by GeminPatel** [zendesk/racecar#53]
When using start_from_beginning: true there is an issue. Once the complete queue is consumed, it starts consumption from the very start. This goes on over and over.

Logs:

```
D, [2018-01-04T17:16:01.990400 #97361] DEBUG -- : Marking citruspay_usr_cpuser_detail/1:8409 as processed
D, [2018-01-04T17:16:01.990637 #97361] DEBUG -- : Fetching batch from citruspay_usr_cpuser_detail/1 starting at offset 8410
D, [2018-01-04T17:16:01.990790 #97361] DEBUG -- : Sending fetch API request 82 to 10.100.93.176:9092
D, [2018-01-04T17:16:01.990966 #97361] DEBUG -- : Waiting for response 82 from 10.100.93.176:9092
D, [2018-01-04T17:16:01.992103 #97361] DEBUG -- : Received response 82 from 10.100.93.176:9092
E, [2018-01-04T17:16:01.992844 #97361] ERROR -- : Invalid offset for citruspay_usr_cpuser_detail/1, resetting to default offset
D, [2018-01-04T17:16:01.993055 #97361] DEBUG -- : Sending list_offset API request 83 to 10.100.93.176:9092
D, [2018-01-04T17:16:01.993245 #97361] DEBUG -- : Waiting for response 83 from 10.100.93.176:9092
D, [2018-01-04T17:16:01.995727 #97361] DEBUG -- : Received response 83 from 10.100.93.176:9092
D, [2018-01-04T17:16:01.995975 #97361] DEBUG -- : Fetching batch from citruspay_usr_cpuser_detail/1 starting at offset 0
D, [2018-01-04T17:16:01.996267 #97361] DEBUG -- : Sending fetch API request 84 to 10.100.93.176:9092
D, [2018-01-04T17:16:01.996970 #97361] DEBUG -- : Waiting for response 84 from 10.100.93.176:9092
D, [2018-01-04T17:16:03.055945 #97361] DEBUG -- : Received response 84 from 10.100.93.176:9092
D, [2018-01-04T17:16:03.110114 #97361] DEBUG -- : Marking citruspay_usr_cpuser_detail/1:26 as processed
I, [2018-01-04T17:16:03.110237 #97361]  INFO -- : Sending heartbeat...
D, [2018-01-04T17:16:03.110406 #97361] DEBUG -- : Sending heartbeat API request 85 to 10.100.93.176:9092
D, [2018-01-04T17:16:03.110886 #97361] DEBUG -- : Waiting for response 85 from 10.100.93.176:9092
D, [2018-01-04T17:16:03.112038 #97361] DEBUG -- : Received response 85 from 10.100.93.176:9092
D, [2018-01-04T17:16:03.122607 #97361] DEBUG -- : Marking citruspay_usr_cpuser_detail/1:27 as processed
```
  
Versions:

```
- Racecar: 0.3.5
- Ruby Kafka: 0.5.1
- Ruby: 2.3.1
```

**Comment by mensfeld** [comment]
@GeminPatel could you provide more details? Ruby version, racecar version, ruby-kafka version, kafka version and so on? Plus some example code as well.

**Comment by dasch** [comment]
@GeminPatel there were some fixes for offset handling in the most recent beta version of ruby-kafka, v0.5.2.beta3 – can you try reproducing the problem with that version?

**Comment by GeminPatel** [comment]
Please check I have updated the post. Code Is exactly the same, minus some name changes

```

class UserConsumer < Racecar::Consumer
  @@bootstrap_started = false
  subscribes_to 'user', start_from_beginning: true

  def initialize
    @row = nil
    @type = nil
  end

  def process(message)
    # not handling JSON.parse failure or empty data as we do not expect it.
    @row = JSON.parse(message.value)
    @type = @row['type']
    Rails.logger.debug("################# #{@type} - #{@row} ##############")
  end
end
```

**Comment by GeminPatel** [comment]
@dasch  Ok Let me try. I will update on the same in some time

**Comment by dasch** [comment]
Note that version 0.5.2.beta3 has the fixes, not 0.5.1.

**Comment by dasch** [comment]
@GeminPatel this should be fixed by v0.5.2 of ruby-kafka.
